src/new/wallet.js

```diff
@@ -2364,34 +2371,21 @@ class HathorWallet extends EventEmitter {
    * }>} Input and signature information
    */
   async getSignatures(tx, { pinCode = null } = {}) {
-    if (await this.storage.isReadonly()) {
+    if (await this.isReadonly()) {
       throw new WalletFromXPubGuard('getSignatures');
     }
     const pin = pinCode || this.pinCode;
-    if (!pin) {
```

Comment:
Why did you remove this?

Reply:
Added the pin check back, but the rest was moved to the new signing method.